### PR TITLE
Add team page member display

### PR DESCRIPTION
**Pull request by cchrischen**
### Summary 
- Implemented the member display section of the team page using the fall 2023 archive json
- Add shadcn card component
- Add types file to use common-types

This pull request is the first step towards implementing the **team page**

- [x] implemented member display
- [x] add responsiveness

### Notion/Figma Link 
https://www.figma.com/file/jKbvR1jVr1k80GGWnCpiRT/Website-Designs-SP23%2FFA23?type=design&node-id=5464-9095&mode=design&t=5jZUX1RMV8eKUdRC-0

https://www.notion.so/Team-Page-Team-Member-Display-5d8845dc798f4145b541c4282c4a1417

### Test Plan 
see video!
https://github.com/cornell-dti/idol/assets/144409850/a4fe1d6a-0f0b-4dc1-b1cd-9fb3f81d7309

### Notes 
- martha.png is a placeholder

### CODE REVIEW

**Comment by oscarwang20** [comment]
Thanks for all the hard work on this Chris! Looks awesome so far.

**Review by alyssayzhang — COMMENTED** [review]
This looks really good 🙌 🙌!!  I'm wondering if we should display Technical PM as Technical Product Manager ?

File: new-dti-website/components/team/MemberGroup.tsx
Comment (henry-li-06) [inline new-dti-website/components/team/MemberGroup.tsx:252]: Ok so I've since learned that using inline arrow functions like this is kinda "frowned upon." It's supposedly less performant (although very marginally) because it needs to be redefined for each re-render but more importantly I think it's a little less readable (although also debatably marginally).  

It's theoretically better to have 
`const mouseHandler = () => setHover(prev => !prev)` and `onMouseEnter={mouseHandler}`

But feel free to ignore, this is just an FYI for the sake of sharing information.

File: new-dti-website/components/team/MemberGroup.tsx
Comment (henry-li-06) [inline new-dti-website/components/team/MemberGroup.tsx:329]: This is a fairly small thing but I'm guessing that `martha.png` was just pulled off Google since the image dimensions aren't very nicely rounded. It probably wouldn't hurt to at least crop the image to whatever the expected size of headshots should be according to the design. 

(I just say this because knowing how projects go, this will probably be forgotten about til the actual headshots get put in after this PR gets merged). 

This repo has an `image-cli` that should be able to crop (and also compress but that isn't particularly important rn) to the desired size. @oscarwang20 should be aware of it if not @andrew032011 definitely should be. 

Now if Martha would just fix her weather machine...

**Review by henry-li-06 — COMMENTED** [review]
Wow this is exciting it looks nice :D 

Left some comments, some are very trivial feel free to ignore those

File: new-dti-website/components/team/MemberGroup.tsx
Comment (henry-li-06) [reply new-dti-website/components/team/MemberGroup.tsx:136]: Or maybe even just a data file tbh that maps the team id to the actual team data. And maybe don't want a default still. I think this should theoretically pull from the team API to make sure things stay up to date 🤔

File: new-dti-website/components/team/MemberDisplay.tsx
Comment (cchrischen) [reply new-dti-website/components/team/MemberDisplay.tsx:18]: Yeah, dividing the two member types would be good for privacy reasons. Members can then edit what they want to display on their profile via IDOL, but that's another feature we can talk about.

The team page has risen some concerns about the current database, which we should resolve first before deciding on using `NovaMember` (which I'm leaning towards). First, we need to keep track of the colleges of each member. Majors should be standardized (CS vs. Computer Science) -- these are important when determining the represented colleges and majors in the About portion of the page. Another issue is how to deal with multiple subroles within a main  role, akin to the advisor discussion on a different PR. That is, how to expand `RoleDescription` or some other field to include advisors, APM, PMM, different lead roles, etc.

**Comment by cchrischen** [comment]
### Commit Changes
* Refactor several dictionaries and lists to `team.json`
* Refactor some utility functions to `utils.ts`
* Add Subteam type to common-types

**Comment by henry-li-06** [comment]
> ### Commit Changes
> * Refactor several dictionaries and lists to `team.json`
> * Refactor some utility functions to `utils.ts`
> * Add Subteam type to common-types

Just for informative purposes, if you split up the single _refactoring and fixing nits_ commit into 3 commits with these exact commit messages, you wouldn't need this comment :D. 

That's kinda the purpose of using commits and having non-trivial commit messages in the first place. You can even selectively stage part of a file to be committed using `git add -p`. 

And besides that, it can also be helpful when reviewing because the reviewer can go commit by commit through fairly isolated changes.

File: new-dti-website/components/team/MemberDisplay.tsx
Comment (henry-li-06) [reply new-dti-website/components/team/MemberDisplay.tsx]: w.r.t. `NovaMember` vs ` IdolMember`, that's really just about whether to use the same data and fields bewteen idol and the dti website. The `NovaMember` type can of course be changed to fit whatever the needs of the new dti website are. 

w.r.t. profile stuff:
- The subroles situation i kinda touched on in #588. Related to #471 as well. I think that's the best way to handle this. 
- In terms of standardizing majors, there's a tradeoff that needs to be considered there. The IDOL profile editor is intended to allow self-servicing of changing things like your major. That's just a free-form text box to type your major in. The only "verification" happens at the stage when an idol admin has to check the diff before approving it. I'm not sure at that stage if rejecting a change is worth it because someone put "CS" instead of "Computer Science." The way to prevent that might be to use a dropdown with all the majors, but that requires maintaining a list of majors which is a pain. From a design perspective, I'm not sure where they stand on that, but from an engineering perspective, it might be fine imo as long as it's not something egregious like the major is misspelled or it's a fake major or something.

File: new-dti-website/components/team/MemberGroup.tsx
Comment (henry-li-06) [inline new-dti-website/components/team/MemberGroup.tsx:90]:
```suggestion
  const mouseHandler = () => setHover(prev => !prev)
```
https://legacy.reactjs.org/docs/hooks-reference.html#functional-updates
https://ogzhanolguncu.com/blog/how-does-prevstate-works-under-the-hood

File: new-dti-website/components/team/MemberGroup.tsx
Comment (andrew032011) [inline new-dti-website/components/team/MemberGroup.tsx:103]: I'd remove this type. We can just set it to a string (Subteam is not a fixed type).

File: new-dti-website/components/team/MemberDisplay.tsx
Comment (andrew032011) [inline new-dti-website/components/team/MemberDisplay.tsx:9]: Delete this type and use `string` instead, since we're reading these values directly from the JSON anyways. So we're not benefitting from any type-checking going on here.

File: new-dti-website/src/app/utils.ts
Comment (andrew032011) [inline new-dti-website/src/app/utils.ts:12]: Instead of the loop, try to do this more functionally with return ` Object.entries(roles).map(...)` instead of editing this in place, which makes it harder to follow (i.e. if this breaks, we'll have a harder time fixing it). Slack me if you need help!

This might involve changing more than just this function, but the refactor will be worth it for code readability.

image

File: new-dti-website/components/team/MemberGroup.tsx
Comment (andrew032011) [inline new-dti-website/components/team/MemberGroup.tsx:274]: Can you leave a comment explaining what `columns` and `index` is and what's going on here in general as well (especially with the checks we're doing underneath)?

Also we shouldn't really be using magic numbers `768` and `1024`, and `2`, `3`, `4`. Instead, let's define consts so that these numbers are self-documenting (i.e. I know what each number means).

Nit: it might be make sense to use a switch/case here instead of if/else.

**Review by andrew032011 — APPROVED** [review]

**Comment by andrew032011** [comment]
Also I'm taking this review over from @henry-li-06 (spoke with him already), so no need to wait for his approval.

**Comment by andrew032011** [comment]
FYI the deploy preview: https://deploy-preview-591--new-cornell-dti.netlify.app/ is crashing now.

**Review by andrew032011 — APPROVED** [review]

**Pull request merged**
